### PR TITLE
Session5

### PR DESCRIPTION
**Pull request by dontcopy**
For review

### CODE REVIEW

File: ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs [hunk 1]
```diff
@@ -12,4 +12,20 @@ public class ItemCreateApiModel
         [Required]
         public string Text { get; set; }
     }
+
+    public class ItemUpdateApiModel
+    {
+        [StringLength(128, MinimumLength = 1)]
+        [Required]
+        public string Text { get; set; }
+        [Required]
```
Comment (jegtugado) [inline ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs:21]: a better practice in REST is to remove the required attribute and get the itemId from the route parameters

**Review by jegtugado — CHANGES_REQUESTED** [review]
Hi sir. Some changes are needed.

Can you also check the ErrorController and remove [HttpGet] from the error action. This particular line may prevent the exception handling for the different Http verbs i.e. post, put, delete

I approved the previous PR so all changes needed will be on this PR. Thanks!

File: ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs (hunk 1 above)
Comment (dontcopy) [reply ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs:21]: Is it redundant to have the itemId in the route parameter and in the model? Can we just have the id in the model or is there a benefit for the recommended approach?

File: ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/Data/ItemService.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        public IEnumerable<ItemDTO> GetAll()
+        {
+            return _mapper.Map<IEnumerable<ItemDTO>>(_repository.All());
+        }
+
+        public IEnumerable<ItemDTO> GetAllByFilter(ItemByFilterDTO filters)
+        {
+            //not enough info on ItemByFilterDto
+            return _mapper.Map<IEnumerable<ItemDTO>>(_repository.All());
```
[the remark this answers is not in the document]
Comment (dontcopy) [reply ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/Data/ItemService.cs:47]: What kind of filter is expected? Will it be like size=3,id=3,text=anything, etc...

[the remark this answers is not in the document]
Comment (jegtugado) [reply ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/Data/ItemService.cs:47]: yes. those are possible values that could be passed

File: ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs (hunk 1 above)
Comment (jegtugado) [reply ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs:21]: if we are to follow rest architecture properly, we don't need the itemId from the model which means you can remove it. Imagine the client sending the id in both the url and body which is redundant

File: ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/Data/ItemService.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jegtugado) [reply ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/Data/ItemService.cs:47]: Hi sir, on session 4 homework there is an example route for the filterBy. i.e. "/items/filterBy?text=text" you can use this as a test case for the filter

File: ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs (hunk 1 above)
Comment (jegtugado) [reply ASPNetCoreMastersTodoList.Api/ASPNetCoreMastersTodoList.Api/ApiModels/ItemApiModel.cs:21]: I also meant to say here that changing this is not required as my comment is a recommendation

**Review by jegtugado — APPROVED** [review]

**Pull request merged**